website/docs/reference/appsmith-architecture/README.md

```diff
@@ -0,0 +1,59 @@
+
+The diagram below illustrates how the different pieces of the platform relate to each other:
+
+![Appsmith Deployment Architecture](/img/Appsmith_Deployment_Architecture.png)
+
+## How it works
+
+A single Docker container runs the following processes to deploy and get started with Appsmith:
+
+1. The **backend server** as a Java process, which handles:
+    1. Authentication: username and password, OAuth 2.0 with Google and GitHub, SSO with OIDC and SAML
+    2. A CRUD API for users, workspaces, applications, pages, widgets
+    3. An action execution module that executes your queries to your databases and APIs
+    4. A git module that maintains clones of your Git-connected apps on the file system.
+
+2. The **NGINX server**, which routes:
+    1. Requests to static assets like Javascript, CSS, and images
+    2. Incoming requests to the backend server, or the RTS depending on the request path
+    3. Path-unidentified requests to an `index.html` page with a 200 status code. This is like a typical single-page application where the client React code is expected to understand the path and show the UI accordingly.
+
+3. The **Real Time Service (RTS)**, a tiny NodeJS server that handles a few features such as mapping dependencies between APIs, queries, and web UI components. It also updates references for all reference-able entities when their names change in Appsmith.
+
+4. The **MongoDB server**, which stores all persistent data needed and generated by Appsmith’s building blocks, like:
+    1. Your users’ details and permissions
+    2. Workspaces you have created
+    3. Applications you are building and have deployed
+    4. Datasources you have connected to
+    5. Queries that connect those datasources to your apps.
+
+5. The **Redis server**, which is used for storing user sessions and cache permissions that a user has access to directly and indirectly.
+
+## Process management
+
+To manage the processes needed to function, Appsmith uses [Supervisor](http://supervisord.org/). This is a lightweight process manager that, when given a list of things to run, runs them, restarts them if they crash, and collects their logs for maintenance and debugging.
+
+When you run the Docker command to start the Appsmith container, it identifies which processes are required to start and run. This includes the backend server, the RTS, and the NGINX server -- as well as any additional processes that haven't been plugged in by the user externally, such as a custom MongoDB or Redis instance.
+
+For each of these processes, the container’s entry-point script (the one that Docker invokes when the container starts) generates a config file with all the details needed for supervisord. For example, the config file generated for the backend Java server includes the full set of commands for starting the process, as well as all necessary environment variables and settings. This same entry-point script also helps set up SSL certificates and install custom CA roots.
```

Comment:
Can we add a cross-reference to the `supervisord`.

Reply:
Added link to appsmith Supervisor page

Reply:
While reading it again, I feel the flow is a bit broken. We are introducing the utility at line 47 but using it before at line 45 too. Let's bring the intro for Supervisord before using it at line 47.

Reply:
Looks like my understanding of the different pieces of Supervisor wasn't great, I looked through its docs some more and updated this section accordingly, I think it might flow a bit better now